### ISSUE TITLE
A major error on the AFCachedImage

### ISSUE THREAD

**Issue by XiangqiTu** [AFNetworking/AFNetworking#3390]
Somethings wrong about calculating imageBytes

Do you want to express this ？

int dataSize = height*bytesPerRow; 

——By your fans

```Objective-C
 self.totalBytes = (UInt64)bytesPerPixel * (UInt64)bytesPerRow;
```

**AFAutoPurgingImageCache.m**
**Line 40**

```Objective-C
@implementation AFCachedImage

-(instancetype)initWithImage:(UIImage *)image identifier:(NSString *)identifier {
    if (self = [self init]) {
        self.image = image;
        self.identifier = identifier;

        CGSize imageSize = CGSizeMake(image.size.width * image.scale, image.size.height * image.scale);
        CGFloat bytesPerPixel = 4.0;
        CGFloat bytesPerRow = imageSize.width * bytesPerPixel;
---
        self.totalBytes = (UInt64)bytesPerPixel * (UInt64)bytesPerRow;
---
        self.lastAccessDate = [NSDate date];
    }
    return self;
}
```

**Comment by kcharwood** [comment]
This is fixed with fac9d87

**Comment by XiangqiTu** [comment]
That’s a long time to receive your response. O(∩_∩)O
